app/models/miq_retire_task.rb

```diff
@@ -78,6 +79,23 @@ def before_ae_starts(_options)
     end
   end
 
+  def lock_retirement
+    lock do
+      reload
+      if retry_retiring?
+        update_attributes(:state => "initializing")
+        _log.info("- creating #{model_being_retired} tasks for #{model_being_retired} <#{self.class.name}:#{id}>")
+        begin
+
```

Comment:
@tinaafitz I have no idea what's supposed to be on line 89.

Reply:
@mkanoor - I was initially thinking the lock should go here, but it seems to me that we should add the lock to the make_retire_request method because we shouldn't create the request if we cant acquire the lock. Can we discuss?